### DISCUSSION TITLE
Branch, Versions and Docs

### DISCUSSION THREAD

**Discussion by npetzall** [schemaspy/schemaspy#805] [Ideas]
**Problem:**
Currently only tags exists for released version. No branches.  
Main/master contains un-released features.

Documentation can be improved for released version. Merg to main/master will not updated the released versions documentation.

Having main/master as default documentation might give the impression that features exist in latest release when they only exist in snapshot.

**Comment by rafalkasa** [comment]
For me we should present current documentation based on the last release. The master/main branch will always contain the latest documentation version for the snapshot release.

In my opinion problem disappear when we will start releasing more frequently :-)

**Reply by npetzall** [reply]
Sounds like what we said 2 years ago :)

**Reply by rafalkasa** [reply]
I'm ready for preparing the next release note, please let me know when we decide to tag current code new release will be created automatically.

**Reply by npetzall** [reply]
So if we start releasing more often there will be no need for branches.

**Reply by rafalkasa** [reply]
I remember that did you work on some improvements into the build and release stuff after we can't use Travis CI anymore

**Reply by rafalkasa** [reply]
> So if we start releasing more often there will be no need for branches.

Exactly this is what I'm thinking in reference to the documentation versions

**Comment by npetzall** [comment]
@rafalkasa
Moving from PR to discussion.

As far as I know, there is no automatic versioning in RTD. It can build tags and branches.

We have the option of creating a branch for each version, for minor versions or major versions. They all come with different pros and cons.

The level of branching will impact the number of merges that might be needed. A fundamental change would need to go into branches main/master, major, minor and patch for instance and it might be multiple patch, multiple minor. 

In reality docs and application might have different release cycles.

6.1.0 is released, if we update the docs should a 6.1.1 be released, no changes have been done to the code and it would be equal to 6.1.0.

Docs update would be merged into main/master and 6.x or 6.1.x.

Only thing I found on RTD was about stable if the branch name resemble a version number
